src/view/com/composer/select-language/SelectPostLanguagesDialog.tsx

```diff
@@ -0,0 +1,418 @@
+import {Language, LANGUAGES, LANGUAGES_MAP_CODE2} from '#/locale/languages'
+import {languageName} from '#/locale/helpers'
+import {MagnifyingGlassIcon} from '#/lib/icons'
+import {msg} from '@lingui/macro'
+import {Text} from '#/components/Typography'
+import {toPostLanguages, useLanguagePrefs} from '#/state/preferences/languages'
+import {Trans} from '@lingui/macro'
+import {useCallback, useMemo} from 'react'
+import {useLanguagePrefsApi} from '#/state/preferences/languages'
+import {useLingui} from '@lingui/react'
+import {View} from 'react-native'
+import * as Dialog from '#/components/Dialog'
+import * as TextField from '#/components/forms/TextField'
+import * as Toggle from '#/components/forms/Toggle'
+import React from 'react'
+
+const WEB_DIALOG_WIDTH = 600
+const MOBILE_DIALOG_WIDTH = '90%'
+
+export function SelectPostLanguagesBtn() {
+  const {_} = useLingui()
+  const langPrefs = useLanguagePrefs()
+  const t = useTheme()
+  const control = Dialog.useDialogControl()
+
+  const onPressMore = useCallback(async () => {
+    if (isNative) {
+      if (Keyboard.isVisible()) {
+        Keyboard.dismiss()
+      }
+    }
+    control.open()
+  }, [control])
+
+  const postLanguagesPref = toPostLanguages(langPrefs.postLanguage)
+
+  return (
+    <>
+      <Button
+        testID="selectLangBtn"
+        onPress={onPressMore}
+        size="small"
+        hitSlop={LANG_DROPDOWN_HITSLOP}
+        label={_(
+          msg({
+            message: `Post language selection`,
+            comment: `Accessibility label for button that opens dialog to choose post language settings`,
+          }),
+        )}
+        accessibilityHint={_(msg`Opens post language settings`)}
+        style={[a.mx_md]}>
+        {postLanguagesPref.length > 0 ? (
+          <Text
+            style={[
+              {color: t.palette.primary_500},
+              a.font_bold,
+              a.text_sm,
+              {maxWidth: 100},
+            ]}
+            numberOfLines={1}>
+            {postLanguagesPref
+              .map(lang => codeToLanguageName(lang, langPrefs.appLanguage))
+              .join(', ')}
+          </Text>
+        ) : (
+          <GlobeIcon size="xs" style={[{color: t.palette.primary_500}]} />
+        )}
+      </Button>
+
+      <LanguageDialog control={control} />
+    </>
+  )
+}
+
+function LanguageDialog({control}: {control: Dialog.DialogControlProps}) {
+  const {gtMobile} = useBreakpoints()
+
+  return (
+    <Dialog.Outer control={control}>
+      <View
+        style={[
+          a.rounded_2xs,
+          web({
+            maxWidth: gtMobile ? MOBILE_DIALOG_WIDTH : WEB_DIALOG_WIDTH,
+            width: gtMobile ? MOBILE_DIALOG_WIDTH : WEB_DIALOG_WIDTH,
+
+            margin: 'auto',
+            alignSelf: 'center',
+          }),
+        ]}>
+        <Dialog.Handle />
+        <PostLanguagesSettingsDialogInner onClose={control.close} />
+      </View>
+    </Dialog.Outer>
+  )
+}
```

Comment:
Adding a View here messes with the layout on web

<img width="1274" height="966" alt="Screenshot 2025-08-26 at 14 09 13" src="https://github.com/user-attachments/assets/4fe5bb28-8c6e-4365-8450-329e383ab4e0" />

Looks like you can just remove it?

```suggestion
function LanguageDialog({control}: {control: Dialog.DialogControlProps}) {
  return (
    <Dialog.Outer control={control}>
      <Dialog.Handle />
      <PostLanguagesSettingsDialogInner onClose={control.close} />
    </Dialog.Outer>
  )
}
}
```

Here's what that looks like:

<img width="1256" height="945" alt="Screenshot 2025-08-26 at 14 17 45" src="https://github.com/user-attachments/assets/902f0c6b-94e2-4740-b725-3d14605b6841" />